Задание для кандидатов. Выполнить ревью кода.sql

```diff
@@ -1 +1,118 @@
-
+create procedure syn.usp_ImportFileCustomerSeasonal
+	@ID_Record int
+AS
+set nocount on
+begin
+	declare @RowCount int = (select count(*) from syn.SA_CustomerSeasonal)
+	declare @ErrorMessage varchar(max)
+
+-- Проверка на корректность загрузки
+	if not exists (
+	select 1
+	from syn.ImportFile as f
+	where f.ID = @ID_Record
+		and f.FlagLoaded = cast(1 as bit)
+	)
+		begin
+			set @ErrorMessage = 'Ошибка при загрузке файла, проверьте корректность данных'
+			raiserror(@ErrorMessage, 3, 1)
+			return
+		end
+
+	-- Чтение из слоя временных данных
+	select
+		c.ID as ID_dbo_Customer
+		,cst.ID as ID_CustomerSystemType
+		,s.ID as ID_Season
+		,cast(cs.DateBegin as date) as DateBegin
+		,cast(cs.DateEnd as date) as DateEnd
+		,c_dist.ID as ID_dbo_CustomerDistributor
+		,cast(isnull(cs.FlagActive, 0) as bit) as FlagActive
+	into #CustomerSeasonal
+	from syn.SA_CustomerSeasonal cs
+		join dbo.Customer as c on c.UID_DS = cs.UID_DS_Customer
+			and c.ID_mapping_DataSource = 1
+		join dbo.Season as s on s.Name = cs.Season
+		join dbo.Customer as c_dist on c_dist.UID_DS = cs.UID_DS_CustomerDistributor
+			and c_dist.ID_mapping_DataSource = 1
+		join syn.CustomerSystemType as cst on cs.CustomerSystemType = cst.Name
+	where try_cast(cs.DateBegin as date) is not null
+		and try_cast(cs.DateEnd as date) is not null
+		and try_cast(isnull(cs.FlagActive, 0) as bit) is not null
+
+	-- Определяем некорректные записи
+	-- Добавляем причину, по которой запись считается некорректной
+	select
+		cs.*
+		,case
+			when c.ID is null then 'UID клиента отсутствует в справочнике "Клиент"'
+			when c_dist.ID is null then 'UID дистрибьютора отсутствует в справочнике "Клиент"'
+			when s.ID is null then 'Сезон отсутствует в справочнике "Сезон"'
+			when cst.ID is null then 'Тип клиента отсутствует в справочнике "Тип клиента"'
+			when try_cast(cs.DateBegin as date) is null then 'Невозможно определить Дату начала'
+			when try_cast(cs.DateEnd as date) is null then 'Невозможно определить Дату окончания'
+			when try_cast(isnull(cs.FlagActive, 0) as bit) is null then 'Невозможно определить Активность'
+		end as Reason
+	into #BadInsertedRows
+	from syn.SA_CustomerSeasonal as cs
+	left join dbo.Customer as c on c.UID_DS = cs.UID_DS_Customer
+		and c.ID_mapping_DataSource = 1
+	left join dbo.Customer as c_dist on c_dist.UID_DS = cs.UID_DS_CustomerDistributor and c_dist.ID_mapping_DataSource = 1

```

Comment:
```suggestion
	left join dbo.Customer as c_dist on c_dist.UID_DS = cs.UID_DS_CustomerDistributor 
		and c_dist.ID_mapping_DataSource = 1

```
16. Дополнительные условия в join переносятся на следующую строку с 1 отступом.